From d4d7eb03b4b3e9b4d696d84fa5f9c20958c6f060 Mon Sep 17 00:00:00 2001
From: Dimitri John Ledkov <dimitri.ledkov@surgut.co.uk>
Date: Fri, 5 Dec 2025 10:58:14 +0000
Subject: [PATCH] boot-modules: add jdk.crypto.ec for compat

Add empty jdk.crypto.ec module in the JRE image. Some 22+ jre's ship
it, and others don't. It is empty module and provides no
functionality. However, 3rd-party jars may list it as required and
thus without this module in the JRE jimage those jars will failt to
work.

To avoid the need to jlink a custom jre, which adds this empty module,
simply add it by default whilst it still exists upstream.
---
 make/conf/module-loader-map.conf | 1 +
 1 file changed, 1 insertion(+)

diff --git a/make/conf/module-loader-map.conf b/make/conf/module-loader-map.conf
index b628bfbf2da..94fd4246a21 100644
--- a/make/conf/module-loader-map.conf
+++ b/make/conf/module-loader-map.conf
@@ -43,6 +43,7 @@ BOOT_MODULES= \
     java.rmi \
     java.security.sasl \
     java.xml \
+    jdk.crypto.ec \
     jdk.incubator.vector \
     jdk.internal.vm.ci \
     jdk.jfr \
-- 
2.51.0

